games/rtype/src/System/EventSystem.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+** EPITECH PROJECT, 2023
+** RType
+** File description:
+** EventSystem.cpp
+*/
+
+#include "EventSystem.hpp"
+
+#include "Components.hpp"
+#include "OtterGraphic.hpp"
+
+namespace Otter::Game::RType::System::Event {
+
+    void PlayerMovementEvent(auto& velocity, int keyPressed)
+    {
+        if (keyPressed == Otter::Graphic::Raylib::RaylibKeyboard::RaylibKey::LEFT)
+            velocity->accelerationDirection.x += -1;
+        if (keyPressed == Otter::Graphic::Raylib::RaylibKeyboard::RaylibKey::RIGHT)
+            velocity->accelerationDirection.x += 1;
+        if (keyPressed == Otter::Graphic::Raylib::RaylibKeyboard::RaylibKey::UP)
+            velocity->accelerationDirection.y += -1;
+        if (keyPressed == Otter::Graphic::Raylib::RaylibKeyboard::RaylibKey::DOWN)
+            velocity->accelerationDirection.y += 1;
+    }
+
+    void PollEvent(Otter::Core::Orchestrator& ref)
```

Comment:
So if i understand the fonctionnement of the poll event system .
Each entity who posses a keyboard component can recieve comoosant ?